markup_tag_evaluation/awesome_align_tokenize.py

```diff
@@ -1,23 +1,34 @@
 #!/usr/bin/env python3
 import argparse
 from io import TextIOWrapper
-from awesome_align.tokenization_bert import BertTokenizer
-from awesome_align.tokenization_utils import PreTrainedTokenizer
+from awesome_align.tokenization_bert import BertTokenizer  # type: ignore
+from awesome_align.tokenization_utils import PreTrainedTokenizer  # type: ignore
+
+UNK_TOKEN = "[UNK]"
 
 
 def write_lines(f: TextIOWrapper, texts: list[str]):
-    for l in texts:
-        f.write(l)
+    for line in texts:
+        f.write(line)
         f.write("\n")
 
 
+def get_unk_tokens(sentence: str, tokenizer: BertTokenizer) -> list[str]:
+    cleaned_sentence_ar = tokenizer.basic_tokenizer.tokenize(
+            sentence, never_split=tokenizer.all_special_tokens)
+    cleaned_sentence = "".join(cleaned_sentence_ar)
+    unk_tokens = [x for x in cleaned_sentence if x not in tokenizer.vocab]
+    return unk_tokens
```

Comment:
Not sure I get what this method is doing. Could you add some explanation/comments?

In `for x in cleaned_sentence` you're iterating over characters, correct? 
So you're essentially iterating over all the original characters, except the whitespaces I'm assuming?